data/dataset.py

```diff
@@ -32,35 +35,40 @@ def __init__(
             n_velocity_bins=n_velocity_bins,
         )
 
-        self.tokenizer_src = Tokenizer(keys=["pitch", "dstart_bin", "duration_bin", "velocity_bin"])
-        self.tokenizer_tgt = VelocityTokenizer()
+        self.tokenizer_src = src_tokenizer
+        self.tokenizer_tgt = tgt_tokenizer
+
+        if samples_path is None:
+            self.samples_path = (
+                f"tmp/datasets/samples-"
+                f"{self.quantizer.n_dstart_bins}-"
+                f"{self.quantizer.n_duration_bins}-"
+                f"{self.quantizer.n_velocity_bins}-"
+                f"{self.split}-"
+                f"{self.tokenizer_src.keys[0]}-{self.tokenizer_src.keys[1]}-vs-"
+                f"{self.tokenizer_tgt.keys[0]}-{self.tokenizer_tgt.keys[1]}.pt"
+            )
+        else:
+            self.samples_path = samples_path
 
         self.dataset = load_dataset(path="roszcz/maestro-v1", split=split)
 
-        self.src_vocab, self.tgt_vocab = self._build_vocab()
+        self.src_vocab, self.tgt_vocab = self.build_vocab()
 
         self.processed_records, self.unprocessed_records = self.load_dataset()
         self.samples = self.load_samples()
 
     def load_samples(self) -> list[tuple[list[int], list[int]]]:
-        path = (
-            f"tmp/datasets/samples-"
-            f"{self.quantizer.n_dstart_bins}-"
-            f"{self.quantizer.n_duration_bins}-"
-            f"{self.quantizer.n_velocity_bins}-"
-            f"{self.split}-"
-            f"bins-to-vel.pt"
-        )
         samples = []
-        if os.path.isfile(path):
-            samples = torch.load(path)
+        if os.path.isfile(self.samples_path):
+            samples = torch.load(self.samples_path)
         else:
             pbar = tqdm(zip(self.processed_records, self.unprocessed_records), total=len(self.processed_records))
 
             print("Tokenizing ... ")
             for processed_record, unprocessed_record in pbar:
                 src_tokens = self.tokenizer_src(processed_record)
-                tgt_tokens = self.tokenizer_tgt(unprocessed_record)
+                tgt_tokens = self.tokenizer_tgt(processed_record)
```

Comment:
Sorry I'm not up to speed - why is both src and tgt made with processed records now?

Reply:
I have rebranded TokenizedMidiDataset to be a parent class, so I could predict high resolution, tokenized midi data, from low resolution, tokenized midi data ( courious as to what I will find) But I do not currently use this parent class, all the functionallity remained the same, as I began using BinsToVelocityDataset.